src/EFCore/Metadata/Internal/EntityType.cs

```diff
@@ -2959,14 +2958,9 @@ private void CheckDiscriminatorProperty(Property? property)
     /// </summary>
     /// <returns>The name of the property that will be used for storing a discriminator value.</returns>
     public virtual string? GetDiscriminatorPropertyName()
-    {
-        if (BaseType != null)
-        {
-            return ((IReadOnlyEntityType)this).GetRootType().GetDiscriminatorPropertyName();
-        }
-
-        return (string?)this[CoreAnnotationNames.DiscriminatorProperty];
-    }
+        => BaseType != null
+            ? ((IReadOnlyEntityType)this).GetRootType().GetDiscriminatorPropertyName()
+            : (string?)this[CoreAnnotationNames.DiscriminatorProperty];
 
```

Comment:
Flip to avoid reverse logic:

```suggestion
        => BaseType is null
            ? (string?)this[CoreAnnotationNames.DiscriminatorProperty]
            : ((IReadOnlyEntityType)this).GetRootType().GetDiscriminatorPropertyName();
```